src/dmd/mars.d

```diff
@@ -2618,22 +2560,61 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
 version (NoMain) {} else
 private void reconcileCommands(ref Param params)
 {
-    static if (TARGET.OSX)
+    if (target.os == Target.OS.OSX)
     {
         params.pic = PIC.pic;
     }
-    static if (TARGET.Linux || TARGET.OSX || TARGET.FreeBSD || TARGET.OpenBSD || TARGET.Solaris || TARGET.DragonFlyBSD)
+    else if (target.os == Target.OS.Windows)
+    {
+        if (params.pic)
+            error(Loc.initial, "`-fPIC` and `-fPIE` cannot be used when targetting windows");
+        if (dmdParams.dwarf)
+            error(Loc.initial, "`-gdwarf` cannot be used when targetting windows");
+        if (params.is64bit)
+            params.mscoff = true;
+    }
+    else if (target.os == Target.OS.DragonFlyBSD)
+    {
+        if (!params.is64bit)
+            error(Loc.initial, "`-m32` is not supported on DragonFlyBSD, it is 64-bit only");
+    }
+    if (target.os != Target.OS.Windows)
+    {
+        if (!params.is64bit && params.mscoff)
+            error(Loc.initial, "`-m32mscoff` can only be used when targetting windows");
+        if (params.mscrtlib)
+            error(Loc.initial, "`-mscrtlib` can only be used when targetting windows");
+    }
+
+    if (target.os & (Target.OS.linux | Target.OS.FreeBSD | Target.OS.OpenBSD | Target.OS.Solaris | Target.OS.DragonFlyBSD))
     {
         if (params.lib && params.dll)
-            error(Loc.initial, "cannot mix -lib and -shared");
+            error(Loc.initial, "cannot mix `-lib` and `-shared`");
+    }
+    if (target.os == Target.OS.Windows)
+    {
+        foreach(b; params.linkswitchIsForCC[])
```

Comment:
Why can't we just check the length ?

Reply:
because `-L` also pushes to params.linkswitchIsForCC, but pushes `false` instead of `true`